### PR TITLE
fix(core): Fix Lint

### LINKED ISSUE #701: clean up lingering lint issues on main
We still have lingering lint issues on main. As part of hardening we need to take the time to get these cleaned up. Most of them are before we had ci checks in place and a more fine tuned golangci lint configuration.

https://github.com/opentdf/platform/actions/workflows/lint-all.yaml?query=branch%3Amain

### PR DESCRIPTION
**Pull request by pbacon-blaber**
Fixes the lint issues in `sdk` and `service` directories.

Passing lint step: https://github.com/opentdf/platform/actions/runs/8913540059

Closes #701

### CODE REVIEW

File: service/policy/subjectmapping/subject_mapping.go [hunk 1]
```diff
@@ -37,7 +42,7 @@ func (s SubjectMappingService) CreateSubjectMapping(ctx context.Context,
 	rsp := &sm.CreateSubjectMappingResponse{}
 	slog.Debug("creating subject mapping")
 
-	sm, err := s.dbClient.CreateSubjectMapping(context.Background(), req)
+	sm, err := s.dbClient.CreateSubjectMapping(ctx, req)
```
Comment (pbacon-blaber) [inline service/policy/subjectmapping/subject_mapping.go:45]: Just want to flag this as it was flagged with lint and I changed the line to use the context passed as an argument, but want to make sure that's what we want here

Comment (dmihalcik-virtru) [reply service/policy/subjectmapping/subject_mapping.go:45]: yes generally you will either pass the same context or a lightly modified version of the context. This allows [functions to be cancelable](https://go.dev/blog/context), a desirable property especially when they call out to other services e.g. a REST endpoint, RPC or even just perform file i/o on large files.

File: service/wellknownconfiguration/wellknown_configuration.go [hunk 2]
```diff
@@ -28,7 +28,10 @@ func RegisterConfiguration(namespace string, config any) error {
 	if _, ok := wellKnownConfiguration[namespace]; ok {
 		return fmt.Errorf("namespace %s configuration already registered", namespace)
 	}
-	wellKnownConfiguration[namespace] = config.(interface{})
+	if _, ok := config.(error); !ok {
+		return fmt.Errorf("config is not of type error")
```
Comment (dmihalcik-virtru) [inline service/wellknownconfiguration/wellknown_configuration.go]: ?

File: service/wellknownconfiguration/wellknown_configuration.go [hunk 3]
```diff
@@ -39,7 +42,10 @@ func NewRegistration() serviceregistry.Registration {
 		ServiceDesc: &wellknown.WellKnownService_ServiceDesc,
 		RegisterFunc: func(_ serviceregistry.RegistrationParams) (any, serviceregistry.HandlerServer) {
 			return &WellKnownService{}, func(ctx context.Context, mux *runtime.ServeMux, server any) error {
-				return wellknown.RegisterWellKnownServiceHandlerServer(ctx, mux, server.(wellknown.WellKnownServiceServer))
+				if srv, ok := server.(wellknown.WellKnownServiceServer); ok {
+					return wellknown.RegisterWellKnownServiceHandlerServer(ctx, mux, srv)
+				}
+				return fmt.Errorf("failed to assert server as WellKnownServiceServer")
```
Comment (dmihalcik-virtru) [inline service/wellknownconfiguration/wellknown_configuration.go:45]:
```suggestion
				return fmt.Errorf("server must be a WellKnownServiceServer")
```

File: service/internal/idpplugin/keycloak_plugin.go [hunk 4]
```diff
@@ -238,8 +240,8 @@ func getKCClient(kcConfig KeyCloakConfig, ctx context.Context) (*KeyCloakConnect
 	return &keycloakConnector, nil
 }
 
-func expandGroup(groupID string, kcConnector *KeyCloakConnector, kcConfig *KeyCloakConfig, ctx context.Context) ([]*gocloak.User, error) {
-	slog.Info("expandGroup invoked: ", groupID, kcConnector, kcConfig.Url, ctx)
+func expandGroup(ctx context.Context, groupID string, kcConnector *KeyCloakConnector, kcConfig *KeyCloakConfig) ([]*gocloak.User, error) {
+	slog.Info("expandGroup invoked: ", groupID, kcConnector, kcConfig.URL, ctx)
```
Comment (dmihalcik-virtru) [inline service/internal/idpplugin/keycloak_plugin.go:244]:
```suggestion
	slog.InfoContext(ctx, "expandGroup invoked", "groupID", groupID, "kccfg", kcConfig.URL)
```

**Review by dmihalcik-virtru — CHANGES_REQUESTED** [review]

**Review by dmihalcik-virtru — DISMISSED** [review]

**Review by strantalis — APPROVED** [review]

File: service/internal/idpplugin/keycloak_plugin.go (hunk 4 above)
Comment (dmihalcik-virtru) [reply service/internal/idpplugin/keycloak_plugin.go:244]: I'll do this in a follow-up

File: service/integration/kas_registry_test.go [hunk 5]
```diff
@@ -14,13 +14,13 @@ import (
...
 
-var nonExistentKasRegistryId = "78909865-8888-9999-9999-000000654321"
+var nonExistentKasRegistryID = "78909865-8888-9999-9999-000000654321"
 
 type KasRegistrySuite struct {
 	suite.Suite
 	f   fixtures.Fixtures
 	db  fixtures.DBInterface
-	ctx context.Context
+	ctx context.Context //nolint:containedctx // context is used in the test suite
```
Comment (jrschumacher) [inline service/integration/kas_registry_test.go:23]: Should we just go lite on the linting for tests and integrations?

**Comment by jrschumacher** [comment]
Should we run a different set of lint rules for all tests including integration? Seems like we could have a good amount of no lint. 

Also curious how you ran this locally. I ran `LINT_OPTIONS="" make lint` earlier this week and didn't see all of these.

**Review by jrschumacher — APPROVED** [review]

File: service/integration/kas_registry_test.go (hunk 5 above)
Comment (strantalis) [reply service/integration/kas_registry_test.go:23]: We can exclude linters with golangci on `_test.go` files. But probably should try  to fix as much as we can in tests for consistency.

https://golangci-lint.run/usage/configuration/#issues-configuration

**Review by strantalis — APPROVED** [review]

**Pull request merged**